CH-679-CH-738 put the password update into a new endpoint

NatalliaBulyha opened the pull request:
jira[endpoint: change password in personal account] https://cozyhometeam.atlassian.net/browse/CH-738?atlOrigin=eyJpIjoiNWI0NTU5MDczMjYyNDA3MzhjYmM3MzUxYmNmYzY2N2IiLCJwIjoiaiJ9

Commit:
CH-679-CH-738 put the password update into a new endpoint

jira[endpoint: change password in personal account] https://cozyhometeam.atlassian.net/browse/CH-738?atlOrigin=eyJpIjoiNWI0NTU5MDczMjYyNDA3MzhjYmM3MzUxYmNmYzY2N2IiLCJwIjoiaiJ9

Commit:
CH-679-CH-738 changed password validation

jira[endpoint: change password in personal account] https://cozyhometeam.atlassian.net/browse/CH-738?atlOrigin=eyJpIjoiNWI0NTU5MDczMjYyNDA3MzhjYmM3MzUxYmNmYzY2N2IiLCJwIjoiaiJ9

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/userservice/controller/UserSecuredController.java:
```diff
@@ -63,7 +64,19 @@ public ResponseEntity<UserInformationResponse> updateUserInfo(@RequestBody @Vali
+    @Operation(summary = "Update user password in personal account")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = SwaggerResponse.Code.CODE_200, description = SwaggerResponse.Message.CODE_200_FOUND_DESCRIPTION) })
+    @Secured({"ROLE_CUSTOMER"})
```
propose to add "ROLE_MANAGER", "ROLE_ADMIN", admin and manager can also have personal account and should have ability to change personal data

NatalliaBulyha replied:
done

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserServiceImpl.java:
```diff
@@ -27,163 +20,174 @@
+
+    @Override
+    public void updateUserPassword(PasswordUpdateRequest passwords, String userId) {
+        User user = userRepository.findByIdAndStatus("6525835b1578af3164f1c4d3", UserStatusE.ACTIVE)
```
"6525835b1578af3164f1c4d3" change to userId

NatalliaBulyha replied:
done

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserServiceImpl.java:
```diff
@@ -27,163 +20,174 @@
+    public void updateUserPassword(PasswordUpdateRequest passwords, String userId) {
+        User user = userRepository.findByIdAndStatus("6525835b1578af3164f1c4d3", UserStatusE.ACTIVE)
+                .orElseThrow(() -> new DataNotFoundException("Unable to update password. User not found."));
+        user.setPassword(encoder.encode("admiN1$2"));
```
"admiN1$2" change to passwords.getNewPassword()

DariaBoo replied:
suppose it was a test code

NatalliaBulyha replied:
done

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserServiceImpl.java:
```diff
@@ -27,163 +20,174 @@
+    @Override
+    public void deleteUser(String email) {
+        User user = userRepository.findByEmailAndStatus(email, UserStatusE.ACTIVE)
+                .orElseThrow(() -> new IllegalArgumentException("Not user found by the email " + email));
```
propose to use the same exceptions: IllegalArgumentException or DataNotFoundException. In this class were used both.

NatalliaBulyha replied:
it's a bug in our logic, so I use custom exceptions. I will use DataNotFoundException

Commit:
CH-679-CH-738 corrections as noted

jira[endpoint: change password in personal account] https://cozyhometeam.atlassian.net/browse/CH-738?atlOrigin=eyJpIjoiNWY3Y2E2ZTU0ZmIwNDRjZmI4YTYzZTQ1YTdiMDcwMzEiLCJwIjoiaiJ9

Commit:
CH-679-CH-738 corrections as noted

jira[endpoint: change password in personal account] https://cozyhometeam.atlassian.net/browse/CH-738?atlOrigin=eyJpIjoiNWI0NTU5MDczMjYyNDA3MzhjYmM3MzUxYmNmYzY2N2IiLCJwIjoiaiJ9
